### PR TITLE
Fix `RedownloadMediaWorker` not being called on transient S3 failure

### PR DESCRIPTION
**Pull request by ClearlyClaire**

### CODE REVIEW

File: app/lib/activitypub/activity/create.rb [hunk 1]
```diff
@@ -283,6 +283,7 @@ def process_attachments
         RedownloadMediaWorker.perform_in(rand(30..600).seconds, media_attachment.id)
       rescue Seahorse::Client::NetworkingError => e
         Rails.logger.warn "Error storing media attachment: #{e}"
+        RedownloadMediaWorker.perform_in(rand(30..600).seconds, media_attachment.id)
```
Comment (renchap) [inline app/lib/activitypub/activity/create.rb:286]: I would retry more aggressively than this, similar to the Sidekiq retry times? Network errors can be transient, and waiting for 5 minutes (at most) before the next retry the media seems to be a lot?

Comment (ClearlyClaire) [reply app/lib/activitypub/activity/create.rb]: Changed to enqueue immediately. The idea for the randomized wait is to not overwhelm the remote server, but this is less relevant when the cause for the issue is on our end.

**Review by renchap — APPROVED** [review]

**Review by Gargron — APPROVED** [review]

**Comment by ThisIsMissEm** [comment]
Won't this repeat infinitely if S3 is having issues? Shouldn't there be a retry limit passed through?

**Comment by ClearlyClaire** [comment]
> Won't this repeat infinitely if S3 is having issues? Shouldn't there be a retry limit passed through?

No, this just queues a media fetching job that will retry a limited number of times with exponential backoff

**Comment by ThisIsMissEm** [comment]
> > Won't this repeat infinitely if S3 is having issues? Shouldn't there be a retry limit passed through?
> 
> No, this just queues a media fetching job that will retry a limited number of times with exponential backoff

Ooooh, oops, I was reading this as if it was in a worker or service, oops 😅

**Pull request merged**